[CI] SemanticTextFieldMapperTests testAggregatableConsistency failing

Errors also for these tests in the build scan:

```
Tests with failures:	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testDefaults	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testUpdates	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testMinimalToMaximal	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testBoostNotAllowed	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testIndexTimeFieldData	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testUpdatesToModelIdNotSupported	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testMeta	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testIgnoreMalformedTrue	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testSupportsParsingObject	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testCannotBeUsedInMultiFields	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testFetch	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testDeprecatedBoostWarning	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testMinimalSerializesToItself	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testExistsQueryMinimalMapping	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testFetchMany	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testNullInput	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testModelIdNotPresent	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testTextSearchInfoConsistency	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testMinimalIsInvalidInRoutingPath	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testIgnoreMalformedExplicitlyFalse	
 - org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testAggregatableConsistency
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/3zoafn2766sqk/tests/:x-pack:plugin:ml:test/org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests/testAggregatableConsistency


**Reproduction line:**
```
./gradlew ':x-pack:plugin:ml:test' --tests "org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests.testAggregatableConsistency" -Dtests.seed=B5F50D73575A4104 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=be-BY -Dtests.timezone=Asia/Srednekolymsk -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests#testAggregatableConsistency`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testAggregatableConsistency'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.ml.mapper.SemanticTextFieldMapperTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: No handler for type [semantic_text] declared on field [field]

  at __randomizedtesting.SeedInfo.seed([B5F50D73575A4104:26B3130B1BD88764]:0)
  at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:557)
  at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:519)
  at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:515)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:320)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:302)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:201)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:187)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:159)
  at org.elasticsearch.index.mapper.MapperTestCase.testAggregatableConsistency(MapperTestCase.java:152)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: org.elasticsearch.index.mapper.MapperParsingException: No handler for type [semantic_text] declared on field [field]

    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:318)
    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:240)
    at org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:426)
    at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:116)
    at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:555)
    at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:519)
    at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:515)
    at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:320)
    at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:302)
    at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:201)
    at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:187)
    at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:159)
    at org.elasticsearch.index.mapper.MapperTestCase.testAggregatableConsistency(MapperTestCase.java:152)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/ml/build.gradle
+++ b/x-pack/plugin/ml/build.gradle
@@ -1,5 +1,6 @@
 import org.elasticsearch.gradle.VersionProperties
 import org.elasticsearch.gradle.internal.dra.DraResolvePlugin
+import org.elasticsearch.gradle.internal.info.BuildParams
 
 apply plugin: 'elasticsearch.internal-es-plugin'
 apply plugin: 'elasticsearch.internal-cluster-test'
@@ -111,6 +112,12 @@ dependencies {
 artifacts {
   // normal es plugins do not publish the jar but we need to since users need it for extensions
   archives tasks.named("jar")
+}
+
+if (BuildParams.isSnapshotBuild() == false) {
+  tasks.named("test").configure {
+    systemProperty 'es.semantic_text_feature_flag_enabled', 'true'
+  }
 }
 
 tasks.register("extractNativeLicenses", Copy) {

